OrbitClientData/TracepointData.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitClientData/TracepointData.h"
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadConstants.h"
+
+using orbit_client_protos::TracepointEventInfo;
+
+void TracepointData::EmplaceTracepointEvent(uint64_t time, uint64_t tracepoint_hash,
+                                            int32_t process_id, int32_t thread_id, int32_t cpu,
+                                            bool is_same_pid_as_target) {
+  absl::MutexLock lock(&mutex_);
+  num_total_tracepoints_++;
+
+  orbit_client_protos::TracepointEventInfo event;
+  event.set_time(time);
+  CHECK(HasTracepointKey(tracepoint_hash));
+  event.set_tracepoint_info_key(tracepoint_hash);
+  event.set_tid(thread_id);
+  event.set_pid(process_id);
+  event.set_cpu(cpu);
+
+  int32_t insertion_thread_id =
+      (is_same_pid_as_target) ? thread_id : orbit_base::kNotTargetProcessThreadId;
+
+  auto [event_map_iterator, unused_inserted] =
```

Comment:
we decided not to use _ here?

Reply:
We had a lengthy discussion in our daily some time ago (might be during your vacation). The result was "use unused_ prefix". The main reason for not using "_" was that it will still bind it to a name. So for nested loops etc. it can't be reused.